src/connection_string.ts

```diff
@@ -57,11 +57,11 @@ function matchesParentDomain(srvAddress: string, parentDomain: string): boolean
  */
 export function resolveSRVRecord(options: MongoOptions, callback: Callback<HostAddress[]>): void {
   if (typeof options.srvHost !== 'string') {
-    return callback(new MongoParseError('Cannot resolve empty srv string'));
+    return callback(new MongoURIError('Option "srvHost" must not be empty'));
```

Comment:
So, the mongo invalid argument error extends the mongo api error, but the mongo uri error extends the mongo runtime error, even though both are effectively validating arguments. Can we think of a better way to set up this hierarchy? It feels like the uri errors should subclass from the api errors, but maybe there's something better

Reply:
Recapping what we spoke about on Zoom:

For the time being, use the predefined clear-cut errors where appropriate. Otherwise default to MongoAPI and MongoRuntime to differentiate between "user's fault" and "driver's fault" errors. We'll sort out instances of MongoAPI and MongoRuntime in another PR.